pkg/policy/validate.go

```diff
@@ -951,6 +959,42 @@ func validateResources(rule kyverno.Rule) (string, error) {
 	return "", nil
 }
 
+func validateMatchRequestTypes(rt kyverno.MatchResources) (string, error) {
+	for i := 0; i < len(rt.RequestTypes); i++ {
+		if (reflect.TypeOf(rt.RequestTypes[i]).Kind()) != reflect.String {
+			return fmt.Sprint("Type: ", reflect.TypeOf(rt.RequestTypes[i]).Kind()), fmt.Errorf("only string type of values are allowed in feild 'requestTypes'")
+		}
+		valuesAllowed := map[string]bool{
+			"CREATE":  true,
+			"UPDATE":  true,
+			"DELETE":  true,
+			"CONNECT": true,
+		}
+		if !valuesAllowed[rt.RequestTypes[i]] {
+			return fmt.Sprint("requestTypes: ", rt.RequestTypes[i]), fmt.Errorf("unknown value '%s' found under the 'requestTypes' field. Only the following values are allowed: [CREATE, UPDATE, DELETE, CONNECT]", rt.RequestTypes[i])
+		}
+	}
+	return "", nil
+}
+
+func validateExcludeRequestTypes(rt kyverno.ExcludeResources) (string, error) {
+	for i := 0; i < len(rt.RequestTypes); i++ {
+		if (reflect.TypeOf(rt.RequestTypes[i]).Kind()) != reflect.String {
+			return fmt.Sprint("Type: ", reflect.TypeOf(rt.RequestTypes[i]).Kind()), fmt.Errorf("only string type of values are allowed in feild 'requestTypes'")
+		}
+		valuesAllowed := map[string]bool{
+			"CREATE":  true,
+			"UPDATE":  true,
+			"DELETE":  true,
+			"CONNECT": true,
+		}
+		if !valuesAllowed[rt.RequestTypes[i]] {
+			return fmt.Sprint("requestTypes: ", rt.RequestTypes[i]), fmt.Errorf("unknown value '%s' found under the 'requestTypes' field. Only the following values are allowed: [CREATE, UPDATE, DELETE, CONNECT]", rt.RequestTypes[i])
+		}
+	}
+	return "", nil
+}
+
```

Comment:
It will good if we can we add unit tests for these functions.

Reply:
working on it.